components/script/script_thread.rs

```diff
@@ -574,6 +576,17 @@ impl ScriptThreadFactory for ScriptThread {
 }
 
 impl ScriptThread {
+    pub fn note_newly_transitioning_nodes(nodes: Vec<UntrustedNodeAddress>) {
```

Comment:
Anything that has a transition starting during layout is part of the DOM, therefore cannot be GCed. There's no need for this to be unsafe.